src/TransformOperationExecutor.ts

```diff
@@ -407,10 +411,18 @@ export class TransformOperationExecutor {
 
 }
 
-function instantiateArrayType (arrayType: Function): Array<any> | Set<any> {
+function instantiateArrayType(arrayType: Function): Array<any> | Set<any> {
     const array = new (arrayType as any)();
     if (!(array instanceof Set) && !("push" in array)) {
         return [];
     }
     return array;
 }
+
+export function testForBuffer(): boolean {
+    try {
```

Comment:
We have to keep a eye on performance here. In C# its very expensive to gather Errors. But i don't know how it is in JS/ TS.